lib/vsteam-lib/BaseCompleter.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Management.Automation;
+using System.Management.Automation.Abstractions;
+using System.Management.Automation.Language;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("vsteam-lib.Test")]
+
+namespace vsteam_lib
+{
+   public abstract class BaseCompleter : IArgumentCompleter
+   {
+      protected readonly IPowerShell _powerShell;
+
+      /// <summary>
+      /// This constructor is used during unit testings
+      /// </summary>
+      /// <param name="powerShell">fake instance of IPowerShell used for testing</param>
+      protected BaseCompleter(IPowerShell powerShell) { this._powerShell = powerShell; }
+
+      /// <summary>
+      /// This constructor is used when running in a PowerShell session. It cannot be
+      /// loaded in a unit test.
+      /// </summary>
+      [ExcludeFromCodeCoverage]
+      protected BaseCompleter() : this(new PowerShellWrapper(RunspaceMode.CurrentRunspace)) { }
```

Comment:
Is line 22 and 29 kind of a work around because of the limitations of PowerShell or is this a completely valid solution even if it were not PowerShell based?

Reply:
This is so I can use dependency injection to make it easier to tests. You can't create a PowerShellWrapper with CurrentRunspace during unit tests. Please having an interface makes it easy for me to pass in a fake during testing.